lumigator/python/mzai/backend/backend/services/jobs.py

```diff
@@ -69,7 +70,7 @@ def _get_results_s3_key(self, job_id: UUID) -> str:
             / settings.S3_JOB_RESULTS_FILENAME.format(job_name=record.name, job_id=record.id)
         )
 
-    def create_inference_job(self, request: JobCreate) -> JobResponse:
+    def create_inference_job(self, request: JobInferenceCreate) -> JobResponse:
```

Comment:
I 100% agree with having some common code to deal with jobs!

I also agree about not putting the large change in this PR 😄, but I am happy to start factoring out the common code. Just a heads up though that we have [another PR dual to this](https://github.com/mozilla-ai/lumigator/pull/333) going on and it might help testing the two in isolation first, then using the common method.